cmake/Modules/FindClangFormat.cmake

```diff
@@ -0,0 +1,91 @@
+#
+# .rst: FindClangFormat
+# ---------------
+#
+# The module defines the following variables
+#
+# ``CLANG_FORMAT_EXECUTABLE`` Path to clang-format executable
+# ``CLANG_FORMAT_FOUND`` True if the clang-format executable was found.
+# ``CLANG_FORMAT_VERSION`` The version of clang-format found
+# ``CLANG_FORMAT_VERSION_MAJOR`` The clang-format major version if specified, 0
+# otherwise ``CLANG_FORMAT_VERSION_MINOR`` The clang-format minor version if
+# specified, 0 otherwise ``CLANG_FORMAT_VERSION_PATCH`` The clang-format patch
+# version if specified, 0 otherwise ``CLANG_FORMAT_VERSION_COUNT`` Number of
+# version components reported by clang-format
+#
+# Example usage:
+#
+# .. code-block:: cmake
+#
+# find_package(ClangFormat) if(CLANG_FORMAT_FOUND) message("clang-format
+# executable found: ${CLANG_FORMAT_EXECUTABLE}\n" "version:
+# ${CLANG_FORMAT_VERSION}") endif()
+
+find_program(CLANG_FORMAT_EXECUTABLE
+             NAMES clang-format
+                   clang-format-9
+                   clang-format-9.0
+                   clang-format-8
+                   clang-format-8.0
+                   clang-format-7
+                   clang-format-7.0
+                   clang-format-6.0
+                   clang-format-5.0
+                   clang-format-4.0
+                   clang-format-3.9
+                   clang-format-3.8
+                   clang-format-3.7
+                   clang-format-3.6
+                   clang-format-3.5
+                   clang-format-3.4
+                   clang-format-3.3
```

Comment:
This explicit list is almost begging for some sort of regex. I've thought a little into the problem but found nothing which wouldn't require reimplementing a custom `find_program` :/

Reply:
If you find some solution, let me know. It'll be mighty helpful